services/QuillLMS/app/services/classlink_integration/user_persister.rb

```diff
@@ -83,17 +76,22 @@ def self.creatable_user(user_hash) = user_hash.except(:sourced_id, :given_name,
 
     def self.updatable_user(user_hash) = creatable_user(user_hash).except(:username)
 
-    private def migrate_to_new_instance
-      ClasslinkAccount.find_by(user_id: existing_user.id).update!(classlink_instance_id:)
-      existing_user
+    private def check_for_email_conflict
+      return unless incoming_user[:email] != existing_user&.email
+
+      return unless User.find_by_email(incoming_user[:email])
```

Comment:
Take or leave this: since we only care about existence, `User.exists?(email: incoming_user[:email])` will fetch smaller pay load from the database.  This is so small that it probably never matters, but I figured I'd mention it since it's slightly clearer (to me) that `exists?` doesn't care about the content.